stepper.c

```diff
@@ -61,54 +63,37 @@ static const uint8_t scs_pin_lookup[4] = {
-// Format: MSB, LSB
-static const uint8_t stepper_init_registers[4][18] = {
-  {
-    //XTABLE
+/*
+  Initialization values for spi stepper drivers
+  Format: MSB, LSB
+*/
+uint8_t stepper_init_base[] = {
     0x0C, 0x11, // CTRL,   DTIME=11, ISGAIN=00, EXSTALL=0, MODE=0010, RSTEP=0, RDIR=0, ENBL=1
     0x11, 0xFF, // TORQUE, SMPLTH=001, TORQUE=0xFF
     0x20, 0x30, // OFF,    PWMMODE=0, TOFF=0x30
     0x30, 0x80, // BLANK,  ABT=0, TBLANK=0x80
     0x41, 0x10, // DECAY,  DECMOD=001, TDECAY=0x10
     0x50, 0x40, // STALL,  VDIV=00, SDCNT=00, SDTHR=0x40
     0x6A, 0x59, // DRIVE,  IDRIVEP=10, IDRIVEN=10, TDRIVEP=01, TDRIVEN=01, OCPDEG=10, OCPTH=01
-    0x70, 0x00  // STATUS, init all status flags to 0 
-  },
-  {
-    //YTABLE
-    0x0C, 0x11, // CTRL,   DTIME=11, ISGAIN=00, EXSTALL=0, MODE=0010, RSTEP=0, RDIR=0, ENBL=1  
-    0x11, 0xFF, // TORQUE, SMPLTH=001, TORQUE=0xFF
-    0x20, 0x30, // OFF,    PWMMODE=0, TOFF=0x30
-    0x30, 0x80, // BLANK,  ABT=0, TBLANK=0x80
-    0x41, 0x10, // DECAY,  DECMOD=001, TDECAY=0x10
-    0x50, 0x40, // STALL,  VDIV=00, SDCNT=00, SDTHR=0x40
-    0x6A, 0x59, // DRIVE,  IDRIVEP=10, IDRIVEN=10, TDRIVEP=01, TDRIVEN=01, OCPDEG=10, OCPTH=01
-    0x70, 0x00  // STATUS, init all status flags to 0 
-  },
-  {
-    //GRIPPER
-    0x0C, 0x11, //  CTRL,   DTIME=11, ISGAIN=00, EXSTALL=0, MODE=0010, RSTEP=0, RDIR=0, ENBL=1  
-    0x11, 0xFF, //  TORQUE, SMPLTH=001, TORQUE=0xFF
-    0x20, 0x30, //  OFF,    PWMMODE=0, TOFF=0x30
-    0x30, 0x80, //  BLANK,  ABT=0, TBLANK=0x80
-    0x41, 0x10, //  DECAY,  DECMOD=001, TDECAY=0x10
-    0x50, 0x40, //  STALL,  VDIV=00, SDCNT=00, SDTHR=0x40
-    0x6A, 0x59, //  DRIVE,  IDRIVEP=10, IDRIVEN=10, TDRIVEP=01, TDRIVEN=01, OCPDEG=10, OCPTH=01   
-    0x70, 0x00  //  STATUS, init all status flags to 0 
-
-  },
-  {
-    //CAROUSEL
-    0x0C, 0x11, //  CTRL,   DTIME=11, ISGAIN=00, EXSTALL=0, MODE=0010, RSTEP=0, RDIR=0, ENBL=1  
-    0x11, 0xFF, //  TORQUE, SMPLTH=001, TORQUE=0xFF
-    0x20, 0x30, //  OFF,    PWMMODE=0, TOFF=0x30
-    0x30, 0x80, //  BLANK,  ABT=0, TBLANK=0x80
-    0x41, 0x10, //  DECAY,  DECMOD=001, TDECAY=0x10
-    0x50, 0x40, //  STALL,  VDIV=00, SDCNT=00, SDTHR=0x40
-    0x6A, 0x59, //  DRIVE,  IDRIVEP=10, IDRIVEN=10, TDRIVEP=01, TDRIVEN=01, OCPDEG=10, OCPTH=01   
-    0x70, 0x00  //  STATUS, init all status flags to 0 
-  }
+    0x70, 0x00  // STATUS, init all status flags to 0
+};
+
+struct spi_stepper_base_diff {
+  uint8_t num_diffs;
+  uint8_t * diffs;
+};
+
+static uint8_t xtable_diffs[] = {0};
+static uint8_t ytable_diffs[] = {0};
+static uint8_t gripper_diffs[] = {0};
+static uint8_t carousel_diffs[] = {0};
+
+static const struct spi_stepper_base_diff spi_stepper_diff_list[4] =
+{
+  {.num_diffs = ARRAY_SIZE(xtable_diffs), .diffs = xtable_diffs},
+  {.num_diffs = ARRAY_SIZE(ytable_diffs), .diffs = ytable_diffs},
+  {.num_diffs = ARRAY_SIZE(gripper_diffs), .diffs = gripper_diffs},
+  {.num_diffs = ARRAY_SIZE(carousel_diffs), .diffs = carousel_diffs}
```

Comment:
A couple of things:

1) num_diffs should be ARRAY_SIZE(x) / 2 (since each overwrite will be an address+value pair)
2) as-written, this is incorrect since you really want num_diff to be 0 currently.